src/kOS.Safe/Encapsulation/Structure.cs

```diff
@@ -26,7 +29,26 @@ static Structure()
         protected Structure()
         {
             instanceSuffixes = new Dictionary<string, ISuffix>(StringComparer.OrdinalIgnoreCase);
-            InitializeInstanceSuffixes();
+            RegisterInitializer(InitializeInstanceSuffixes);
+        }
+
+        protected void RegisterInitializer(Action callback)
+        {
+            if (!initializeSuffixCallbacks.Contains(callback))
+            {
+                initializeSuffixCallbacks.Add(callback);
+            }
+        }
+
+        private void callInitializeSuffixes()
+        {
+            if (needToInitializeSuffixes)
+            {
+                foreach (var callback in initializeSuffixCallbacks)
+                {
+                    callback.Invoke();
+                }
+            }
```

Comment:
I like the system you're setting up here.  As long as we still also have an escape route to do it per-instance for those few suffixes that need it (like the resources in a fuel tank).